internal/pkg/consensus/genesis.go

```diff
@@ -188,15 +195,7 @@ func MakeGenesisFunc(opts ...GenOption) GenesisInitFunc {
 		for addr, val := range genCfg.miners {
 			a := miner.NewActor()
 			a.Balance = val.balance
-			s := storageMap.NewStorage(addr, a)
-			scid, err := s.Put(val.state)
-			if err != nil {
-				return nil, err
-			}
-			if err = s.LegacyCommit(scid, a.Head); err != nil {
-				return nil, err
-			}
-			if err := st.SetActor(ctx, addr, a); err != nil {
+			if err := st.SetActor(context.Background(), addr, a); err != nil {
```

Comment:
I think that this change is you regressing genesis behavior of actually setting up miner actors as specified in the genCfg in order to move to the new state machine.  This is fine, but you should leave a Dragons saying: "actually write miner actor storage" or something so that we don't have to track this down later.

Reply:
Reading your `createActor` in `SetupDefaultActors` function below I'm a little confused why you don't abstract that to a top level helper and then run it here on the miner actor state from the genesis config.

Reply:
Short answer, most of this file needs to get re-written once we import the new actor code.